mtags-java/src/main/scala/scala/meta/internal/pc/JavaKeyword.scala

```diff
@@ -0,0 +1,35 @@
+package scala.meta.internal.pc
+
+case class JavaKeyword(
+    name: String,
+    level: JavaKeyword.Level
+)
+
+object JavaKeyword {
+  sealed trait Level
+
+  case object TopLevel extends Level
+  case object ClassLevel extends Level
+  case object MethodLevel extends Level
+
+  val topLevelKeywords: List[JavaKeyword] = List(
+    "package", "import", "public", "private", "protected", "abstract", "class",
+    "interface"
+  ).map(JavaKeyword(_, TopLevel))
+
+  val classLevelKeywords: List[JavaKeyword] = List(
+    "public", "private", "protected", "static", "final", "native",
+    "synchronized", "abstract", "default", "class", "interface", "void",
+    "boolean", "int", "long", "float", "double", "extends", "implements"
+  ).map(JavaKeyword(_, ClassLevel))
+
+  val methodLevelKeywords: List[JavaKeyword] = List(
```

Comment:
`char`, `byte` and `short` might be missing from this list. :slightly_smiling_face: